02-frequency_filter.py

```diff
@@ -63,34 +75,53 @@ def run_filter(subject, run=None, session=None):
     raw.load_data()
 
     # Band-pass the data channels (MEG and EEG)
-    msg = f'Filtering data between {config.l_freq} and {config.h_freq} (Hz)'
+    msg = (f'Filtering experimental data between {config.l_freq} and '
+           f'{config.h_freq} Hz')
     logger.info(gen_log_message(message=msg, step=2, subject=subject,
-                                session=session, run=run,))
+                                session=session, run=run))
 
-    raw.filter(config.l_freq, config.h_freq,
-               l_trans_bandwidth=config.l_trans_bandwidth,
-               h_trans_bandwidth=config.h_trans_bandwidth,
-               filter_length='auto', phase='zero', fir_window='hamming',
-               fir_design='firwin'
-               )
+    filter_kws = dict(l_freq=config.l_freq, h_freq=config.h_freq,
+                      l_trans_bandwidth=config.l_trans_bandwidth,
+                      h_trans_bandwidth=config.h_trans_bandwidth,
+                      filter_length='auto', phase='zero', fir_window='hamming',
+                      fir_design='firwin')
+    raw.filter(**filter_kws)
+
+    if config.noise_cov == 'emptyroom':
+        msg = f'Filtering empty-room recording.'
+        logger.info(gen_log_message(message=msg, step=2, subject=subject,
+                                    session=session, run=run,))
+        raw_er = mne.io.read_raw_fif(raw_er_fname_in)
+        raw_er.load_data()
+        raw_er.filter(**filter_kws)
```

Comment:
Just out of curiosity, why do you need this branching code path? Why not simply call the function with `subject='emptyroom'` ?

Reply:
Because then you wouldn't know whose subject's empty-room measurement to process :)

Reply:
humm ... but all empty-rooms are measured similarly, no? Except the bad channel part

Reply:
But the recordings themselves may be different. Typically there is one per day. Now two participants measured on two separate days will also have separate ER recordings that match their experimental session.... or maybe I don’t really understand your question :)

Reply:
humm ... why can't you simply say:

```py
sessions = glob.glob('some globbing magic here'))
parallel(run_func(subject='emptyroom', session=session) for session in sessions)
```

Reply:
Basically I don't see a reason why the emptyroom should be processed in relation with a particular file because all files are processed the same way.

Reply:
You mean simply grabbing all empty-room recordings and processing them?

- We could end up processing recordings we don't want to analyze. Assume you have a dataset with 600 participants, recorded over 200 days, but only want to analyze 10 of them – you wouldn't want to process 200 empty-room recordings!

- We need to ensure we set the same channels as bad as in the matching participant recording

Reply:
Also there could be >1 empty-room recording per day, or sometimes a recording could be missing for the specific day. You need to find the one that's closest in time to the participant's recording. It's just easier to do that while you're handling that particular participant's data anyway

Reply:
okay fair enough. I was going to suggest that you could get the list of matched empty room dates first. But maybe it's the same effort and code complexity

> Also there could be >1 empty-room recording per day, or sometimes a recording could be missing for the specific day. 

I think it's good enough if you find a recording for that day.